### PR TITLE
Add a ggui version for `demo_3d`

### PR DESCRIPTION
**Pull request by Jack12xl**
It will remain a draft pull request until the `Taichi==0.8.9` is released.
Then I will reverse some temporary changes due to the `ti.static` bug.

### CODE REVIEW

**Comment by Jack12xl** [comment]
ggui_demo_v0002

**Comment by Jack12xl** [comment]
ggui_demo_v0004

**Comment by yuanming-hu** [comment]
Cool, thanks. Could you please pick a better camera position?

File: engine/mpm_solver.py [hunk 1]
```diff
@@ -130,6 +130,7 @@ def __init__(
             self.material = ti.field(dtype=ti.i32)
         # Particle color
         self.color = ti.field(dtype=ti.i32)
+        self.color_with_alpha = ti.Vector.field(4, dtype=ti.f32)
```
Comment (yuanming-hu) [inline engine/mpm_solver.py:133]: Shall we add a `ggui_support=False` flag so that this field is allocated only if we use GGUI?.

**Review by yuanming-hu — COMMENTED** [review]
Very cool!

File: engine/mpm_solver.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 USE_IN_BLENDER = False
 
-ti.require_version(0, 8, 1)
+# ti.require_version(0, 8, 8)
```
Comment (Jack12xl) [inline engine/mpm_solver.py:10]: Didn't find a equivlent function in `0.8.9`

File: README.md [hunk 3]
```diff
@@ -14,8 +14,12 @@
  - Execute `python3 demo/demo_2d.py` and you will see
 
 <img src="https://github.com/yuanming-hu/public_files/raw/master/graphics/elements/demo_2d.gif">
- 
+
  - Execute `python3 demo_3d.py` and you will see a 3D simulation visualized in 2D
+ - Execute `python3 demo_3d_ggui.py` and you will see a 3D simulation rendered with the [GGUI](https://docs.taichi.graphics/lang/articles/misc/ggui).  Make sure to install the [Vulkan Support](https://developer.nvidia.com/vulkan-driver) first.
```
Comment (yuanming-hu) [inline README.md:19]:
```suggestion
 - Execute `python3 demo_3d_ggui.py` and you will see a 3D simulation rendered with [GGUI](https://docs.taichi.graphics/lang/articles/misc/ggui). Note that GGUI requires Vulkan so please make sure your platform supports that.
```

File: README.md [hunk 4]
```diff
@@ -36,8 +40,8 @@ python3 engine/render_particles.py \
      -M 460 \
      --shutter-time 0.0 \
      -r 128
-```
-   
+ ```
```
Comment (yuanming-hu) [inline README.md:43]:
```suggestion
```
```

File: engine/mpm_solver.py (hunk 2 above)
Comment (yuanming-hu) [reply engine/mpm_solver.py:10]: That's moved to `ti.init`. Let's simply skip this check as you already have that in `requirements.txt`.

Comment (yuanming-hu) [inline engine/mpm_solver.py:10]:
```suggestion
```

**Review by yuanming-hu — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**